### ISSUE TITLE
Regression not effective enough on customized network

### ISSUE THREAD

**Issue by ztianlin** [lucastabelini/LaneATT#83]
Really appreciated for this project! I have modified the network a little bit for some task & performance concerns and what I found was that the predicted lanes were very straight (curves are expected). Could you give me some advices on this ? Thanks in advance.

**Comment by ztianlin** [comment]
What I did:
I use 32 1x1 convs to get a [batch_size, 32, fmap_h, fmap_w] feature map after the backbone. After anchors cutting, each anchor feature's size is [batch_size, num_anchors, 32, f_map_h]. Then I feed the first 2 channel of it into a global average pooling to get the classification result directly. For regression, I feed the next 20 channels into a linear layer just like what you did. And for the last 10 channels, I feed them into another global average pooling to get subclasses logits of each lane.

**Comment by ztianlin** [comment]
I have three tasks, lane classification (2 classes), anchors horizontal regression (72 offsets) and lane sub-classification (10 classes, focol loss).  Can you give me some suggestions on the loss weights? I have no idea how to set them.

**Comment by lucastabelini** [comment]
>  I found was that the predicted lanes were very straight (curves are expected). Could you give me some advices on this ?

If the training dataset is not balanced (i.e., most of the lanes are straight), perhaps you could try to balance it by augmenting the curved ones. 

> Can you give me some suggestions on the loss weights? I have no idea how to set them.

You should try to set the weights in a way that all loss components have approximately the same magnitude. Otherwise, the model might focus on the larger component and ignore the others with smaller magnitude.

**Comment by ztianlin** [comment]
Thank you very much! And do you think my average pooling thing is ok compared to the original fully connected heads?

**Comment by lucastabelini** [comment]
Not sure. I would say that using a fully connected layer should probably yield a better result and also should not be too computationally heavy. However, if the results you are getting with the global average pooling are sufficient in your application then I don't see a problem.

**Comment by ztianlin** [comment]
Ok. I will carry on experiments for the heads. Thanks again~
